### PR TITLE
WIP: Use upstream sphinx templates

### PR DESCRIPTION
**Pull request by jayvdb**

### CODE REVIEW

**Comment by jayvdb** [comment]
One part is not backwards compatible: variable `sphinx_extensions`, replacing block `SPHINX_EXTENSIONS`

**Comment by jayvdb** [comment]
Not this depends on https://github.com/moremoban/moban/issues/26

File: templates/conf.py.jj2 [hunk 1]
```diff
@@ -7,50 +7,48 @@ DESCRIPTION = (
...
 )
-extensions = [
-    'sphinx.ext.autodoc',
-    'sphinx.ext.doctest',
-    'sphinx.ext.intersphinx',
-    'sphinx.ext.viewcode',
-{%block SPHINX_EXTENSIONS%}
-{%endblock%}
-]
-{%block custom_varaibles%}
```
Comment (jayvdb) [inline templates/conf.py.jj2]: this is https://github.com/moremoban/pypi-mobans/pull/20

File: templates/conf.py.jj2 [hunk 2]
```diff
@@ -7,50 +7,48 @@ DESCRIPTION = (
...
-{%endblock%}
-intersphinx_mapping = {
```
Comment (jayvdb) [inline templates/conf.py.jj2]: this is the only difference between the two conf.py.jj2 , and it doesnt seem worthwhile having two different templates if this is the only difference.

File: templates/docs/master_doc.rst_t [hunk 3]
```diff
@@ -0,0 +1,21 @@
+.. {{ project }} documentation master file, created by
+   sphinx-quickstart on {{ now }}.
```
Comment (jayvdb) [inline templates/docs/master_doc.rst_t]: The "created by sphinx-quickstart" is a bit annoying, and is an example of how `copy` targets might be modified , here after copy, replace "sphinx-quickstart" with `moban` or `yehua`

`{{ now }}` doesnt exist yet, and is a bad idea for re-generated files.

maybe we can delete these lines by building a slightly different 'include' command.

File: templates/docs/conf.py_t [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+# The name of the Pygments (syntax highlighting) style to use.
+pygments_style = 'sphinx'
+
+
+# -- Options for HTML output -------------------------------------------------
+
+# The theme to use for HTML and HTML Help pages.  See the documentation for
+# a list of builtin themes.
+#
+html_theme = 'alabaster'
```
Comment (chfw) [inline templates/docs/conf.py_t]: `html_theme` overrides previous `html_theme` variable.

Comment (jayvdb) [reply templates/docs/conf.py_t]: The include is first. Then our template overrides the hardcoded value.

Comment (chfw) [reply templates/docs/conf.py_t]: got you

File: .travis.yml [hunk 5]
```diff
@@ -4,6 +4,11 @@ notifications:
   email: false
 install:
   - pip install -r test_requirements.txt
+before_script:
+  - git clone https://github.com/sphinx-doc/sphinx ../sphinx
+  - (cd .. && ln -s sphinx/sphinx/templates/quickstart/ sphinx-quickstart-templates)
+  # Check upstream mobans are in sync
+  - moban -m upstream.yml
```
Comment (chfw) [inline .travis.yml]: I think this command will update sphinx related files in the repo and then the tests will be performed on top of them. Do you imagine that we update them offline? for example, when the build fails, and we try to update those sphinx files, e.g. Make.bat.jj2. How would the update work?

Comment (MacBox7) [reply .travis.yml]: I believe make.bat should not be templated and must be autogenerated. Refer [this](http://www.sphinx-doc.org/en/master/usage/quickstart.html#running-the-build). I believe changes must be done in `setup.py` to autogenerate this.

File: templates/docs/conf.py_t [hunk 6]
```diff
@@ -0,0 +1,209 @@
+# -*- coding: utf-8 -*-
```
[the remark this answers is not in the document]
Comment (jayvdb) [reply templates/docs/conf.py_t]: https://github.com/moremoban/moban/issues/60 and https://github.com/moremoban/moban/issues/36

File: templates/docs/index.rst.jj2 [hunk 7]
```diff
@@ -0,0 +1,13 @@
+{% macro underlines(name) %}
```
Comment (jayvdb) [inline templates/docs/index.rst.jj2]: I think it might be possible to share this macro with `templates/docs/source/index.rst.jj2`

**Comment by jayvdb** [comment]
Before merging, I want to sync a repo with these changes to check what it looks like.

File: templates/docs/source/index.rst.jj2 [hunk 8]
```diff
@@ -0,0 +1,6 @@
+{% macro underlines(name) %}
```
Comment (chfw) [inline templates/docs/source/index.rst.jj2]: what's the purpose of two index.rst.jj2 here?

File: templates/docs/index.rst.jj2 [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+{%- for i in range(name|length) -%}={%- endfor -%}
```
[the remark this answers is not in the document]
Comment (chfw) [reply templates/docs/index.rst.jj2]: have you tried https://github.com/moremoban/moban#split_length?

**Pull request merged**